schema/resourcedefinition.proto

```diff
@@ -60,23 +60,30 @@ enum Type {
   DOUBLE = 4;
   FLOAT = 5;
   BOOLEAN = 6;
-  OBJECT = 7;
-  ARRAY = 8;
 }
 
 message Property {
-  Type type = 1;
+  oneof types {
+    Type type = 1;
+    ObjectType object_type = 5;
+    ArrayType array_type = 6;
+  }
   // field number used for protobuf or other systems where fields must
   // be explicitly enumerated.
   int32 number = 2;
   bool readOnly = 3;
   bool required = 4;
+}
 
-  // If type is OBJECT, this is the name of the object in `messages`.
-  string object_type = 5;
+message ObjectType {
+  oneof object_details {
+    string message_name = 1;
```

Comment:
would ObjectType be better as just a string? or do you have an idea on some additional parameters that an object may want?

Reply:
At some point, we want to have objects that are a list of properties (instead of this message reference thing). This gives us a very clean interface to add a list of properties in the near future.